### PR TITLE
DataNode: Update to OpenSearch 2.11.0

### PR DESCRIPTION
**Pull request by janheise**
## Description

Cloud is already migrating into 2.11.0 so it makes sense to update here also.

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java [hunk 1]
```diff
@@ -46,15 +48,15 @@ void testOpensearch2() throws URISyntaxException {
...
+                    Assertions.assertThat(node.nodeVersion()).isEqualTo(OPENSEARCH_VERSION_IN_DATANODE);
                     Assertions.assertThat(node.indices())
                             .hasSize(6)
                             .extracting(IndexInformation::indexName)
                             .contains(".opensearch-sap-log-types-config", ".plugins-ml-config", "graylog_0", ".opensearch-observability", ".opendistro_security", "security-auditlog-2023.11.24");
 
                     final IndexInformation graylog_0 = node.indices().stream().filter(i -> i.indexName().equals("graylog_0")).findFirst().orElseThrow(() -> new RuntimeException("Failed to detect graylog_0 index"));
 
-                    Assertions.assertThat(graylog_0.indexVersionCreated()).isEqualTo("2.10.0");
+                    Assertions.assertThat(graylog_0.indexVersionCreated()).isEqualTo(OPENSEARCH_VERSION_IN_DATANODE);
```
Comment (todvora) [inline data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java:59]: The index directory has been created by 2.10 and will stay in that version if we won't regenerate the files with a newer one. This is why the test is now failing here.

File: data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java [hunk 2]
```diff
@@ -46,15 +48,15 @@ void testOpensearch2() throws URISyntaxException {
         Assertions.assertThat(result.nodes())
                 .hasSize(1)
                 .allSatisfy(node -> {
-                    Assertions.assertThat(node.nodeVersion()).isEqualTo("2.10.0");
+                    Assertions.assertThat(node.nodeVersion()).isEqualTo(OPENSEARCH_VERSION_IN_DATANODE);
```
Comment (todvora) [inline data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java:51]: The index directory has been created by 2.10 and will stay in that version if we won't regenerate the files with a newer one. This is why the test is now failing here.

File: data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java (hunk 1 above)
Comment (janheise) [reply data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java]: reverted, thx.

File: data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java (hunk 2 above)
Comment (janheise) [reply data-node/src/test/java/org/graylog/datanode/filesystem/index/IndicesDirectoryParserTest.java]: reverted, thx

**Review by todvora — COMMENTED** [review]
LGTM, all tests are green, we are good to merge!

**Review by todvora — APPROVED** [review]

**Pull request merged**